geopandas/plotting.py

```diff
@@ -97,16 +98,18 @@ def plot_polygon_collection(
     if "markersize" in kwargs:
         del kwargs["markersize"]
     if color is not None:
-        kwargs["color"] = color
-        if pd.api.types.is_list_like(color):
-            kwargs["color"] = np.take(color, multiindex)
-        else:
+        if is_color_like(color):
             kwargs["color"] = color
+        else:
+            if pd.api.types.is_list_like(color):
```

Comment:
You can make this into a `elif`

Also wondering what should happen if neither of the checks are True. With the code above, it seems that color  keyword is then ignored (not set into kwargs), but we should maybe raise an error instead?

Reply:
> we should maybe raise an error instead?

I was thinking about the same. I'll add error.